streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBStore.java

```diff
@@ -108,7 +108,9 @@ public class RocksDBStore implements KeyValueStore<Bytes, byte[]>, BatchWritingS
     FlushOptions fOptions;
     private Cache cache;
     private BloomFilter filter;
-    private Statistics statistics;
+
+    // visible for testing
+    protected Statistics statistics;
```

Comment:
I tried to avoid this with mocking the Statistics constructor with no luck. 
However, I managed with reflection.
```
    @Test
    public void shouldCloseStatisticsWhenUserProvidesNoStatistics() throws Exception {
        rocksDBStore = getRocksDBStoreWithRocksDBMetricsRecorder();

        context = getProcessorContext(RecordingLevel.DEBUG);

        rocksDBStore.openDB(context.appConfigs(), context.stateDir());

        Statistics statistics = getStatistics(rocksDBStore);
        rocksDBStore.close();
        assertFalse(statistics.isOwningHandle());
        statistics = getStatistics(rocksDBStore);
        assertNull(statistics);
    }

    private Statistics getStatistics(final RocksDBStore rocksDBStore) throws Exception {
        final Field statisticsField = rocksDBStore.getClass().getDeclaredField("statistics");
        statisticsField.setAccessible(true);
        final Statistics statistics = (Statistics) statisticsField.get(rocksDBStore);
        statisticsField.setAccessible(false);
        return statistics;
    }

```
Normally, I am not a big fan of using reflection, but in this case were we need to verify off-heap memory leaks I think it is fine.

In such a way, we can keep `statistics` as a private field and still verify that there is not memory leak.

WDYT?

Reply:
Done